codalab/rest/bundles.py

```diff
@@ -224,12 +224,16 @@ def _create_bundles():
       "uploading" regardless of the bundle type, or 0 otherwise. Used when
       copying bundles from another CodaLab instance, this prevents these new
       bundles from being executed by the BundleManager. Default is 0.
+    - `parent_worksheet_required`: 0 if the bundle does not need a hosted worksheet
+      (e.g. It is the user's uploaded avatar) or 1 otherwise.
+      Default is 1 since the parent worksheet need to be specified by default.
     """
     worksheet_uuid = request.query.get('worksheet')
     shadow_parent_uuid = request.query.get('shadow')
     after_sort_key = request.query.get('after_sort_key')
     detached = query_get_bool('detached', default=False)
-    if worksheet_uuid is None:
+    parent_worksheet_required = query_get_bool('parent_worksheet_required', default=True)
+    if parent_worksheet_required and worksheet_uuid is None:
```

Comment:
Why not just allow `worksheet_uuid` to be None, in which case the bundle doesn't get added to any worksheet?

Reply:
I did not do so because if we set `worksheet_uuid` to None for avatar bundles, then we have to remove the check of whether the `parent_worksheet` is specified in URL, which may result in some error when creating normal bundles, since the normal bundles still need hosted worksheet to inherit the permission from.